rig/machine_control/machine_controller.py

```diff
@@ -1591,6 +1627,72 @@ def seek(self, n_bytes, from_what=os.SEEK_SET):
             )
 
 
+class _WriteBufferChild(object):
+    def __init__(self, parent):
+        self.parent = parent
+
+    def get_child(self):
+        # Return a sibling
+        return _WriteBufferChild(self.parent)
+
+    def add_new_write(self, *args, **kwargs):
+        self.parent.add_new_write(*args, **kwargs)
+
+    def flush(self):
+        self.parent.flush()
+
+
+class _WriteEntry(collections.namedtuple("_WriteEntry", "start, data")):
+    @property
+    def end(self):
+        return self.start + len(self.data)
+
+
+class _WriteBuffer(_WriteBufferChild):
```

Comment:
I don't think I understand why this needs to be a tree? Why can't everyone just have a reference to the same _WriteBuffer everywhere?